packages/ember-routing/lib/location/hash_location.js

```diff
@@ -106,9 +104,7 @@ export default EmberObject.extend({
     @param callback {Function}
   */
   onUpdateURL(callback) {
-    let guid = guidFor(this);
-
-    jQuery(window).on(`hashchange.ember-location-${guid}`, () => {
+    this._hashchangeHandler = () => {
```

Comment:
it is only called during setup, but I am happy to add code to remove before adding